### ISSUE TITLE
How to turn off type guessing when there are flycheck errors

### ISSUE THREAD

**Issue by bhyland** [chrisdone/intero#585]
I've been using Intero with emacs for the past few weeks, and love most of the functionality it brings out-of-the-box. Thanks for all the hard work.

However, there is one 'feature' that is really annoying me, and I cannot find a related setting, and to be honest I'm not sure if this is a bug, a setup problem, or intentional:

If I have an error somewhere in my code, the type checker starts providing nonsense types, perhaps based on what it infers should go there. A simple example:
```
example :: Int
example =
  let 
      a = 1
      b = 2
      c = "3"
  in
    a + b + c
```

So now, intero shows a redline over 'a', and checking the type of 'a' tells me that it is of type [Char]. I know that the error message gives more information, and in this contrived example it's obvious what's going on. But when I'm trying to fix an error when putting together some complex types from other libraries, those inferred types can completely lead things astray. For example, working on a Yesod app I recently got this:

```
head ::  IO (Either JWTError ClaimsSet)
```
In summary, I want to see the actual types of anything already declared. Assuming what I described here is by design, is there a way I can change this behaviour?  

Thank you.

**Comment by chrisdone** [comment]
In the presence of type errors all type information is from the last available successful type-check, additionally, [defer-type-errors](https://downloads.haskell.org/~ghc/7.8.1/docs/html/users_guide/defer-type-errors.html) is enabled to try to give as much information as possible.

This is intended and not likely to change. When you have a type error you really can't provide too much help because type inference is complicated.

People complained more about missing any information, over getting a few false-positives sometimes. Hence this behavior.
